arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -115,14 +141,31 @@ def getFactsWithDimension(
                     continue
                 elif (fact.context.dimMemberQname(
                         dimensionQn
-                ) == membeQn
+                ) in memberQn
                       and fact.context.qnameDims.keys() == {dimensionQn}):
                     foundFacts.add(fact)
                 elif not len(fact.context.qnameDims):
                     foundFacts.add(fact)
     return foundFacts
 
 
+def getFactsWithoutDimension(
+        val: ValidateXbrl,
+        conceptQn: QName,
+) -> set[ModelFact ]:
+    foundFacts: set[ModelFact] = set()
+    facts = val.modelXbrl.factsByQname.get(conceptQn)
+    if facts:
+        for fact in facts:
+            if fact is not None:
+                if fact.xValid >= VALID:
+                    if fact.context is None:
+                        continue
+                    elif not len(fact.context.qnameDims):
+                        foundFacts.add(fact)
```

Comment:
nit: nesting is a bit high here. IMO a set comprehension is easier to read in situations like this:

```suggestion
    facts = modelXbrl.factsByQname.get(conceptQn, set())
    foundFacts = {
        fact for fact in facts
        if fact is not None
        if fact.xValid >= VALID
        if fact.context is not None
        if len(fact.context.qnameDims) == 0
    }
```